ksql-parser/src/main/java/io/confluent/ksql/parser/tree/CreateStreamAsSelect.java

```diff
@@ -33,15 +33,32 @@
   private final Map<String, Expression> properties;
   private final Optional<Expression> partitionByColumn;
 
-  public CreateStreamAsSelect(NodeLocation location, QualifiedName name, Query query,
-                      boolean notExists, Map<String, Expression> properties,
-                              Optional<Expression> partitionByColumn) {
+  public CreateStreamAsSelect(
```

Comment:
Why do we need to add a constructor that works with a null location? Do we drop it when rewriting? That seems reasonable since the location is the line/character in the original statement. Just want to make sure I understand. But then maybe the rewrite ast should just always drop the location since its likely not going to be correct anymore.

Reply:
The other constructor that accepted Optional is private. I could either make it public or create a new public one. For I went with creating a public one that works with null.
Rewrite will try to keep nodes as the same as original if they don't change.
We can discuss this more later and come up with a way that is constant for all subclasses of the AST visitor.

Reply:
Why do we have two two public constructors, (one taking a location, one the doesn't), which defer to a private constructor taking optional, when this is used from somewhere that has an optional location in the first place?  If the private constructor was public, then the calling code could just pass the optional location, no?  Or am I missing nuance?

Exposing the private constructor on this, (and others like it), would remove the code duplicate in the rewriter.

Reply:
Agree with making the private constructor public to eliminate code duplication. I would postpone this to another PR since it will require changes in many classes not related to this PR.